src/nrniv/nrnbbcore_write.cpp

```diff
@@ -1971,10 +1971,45 @@ int nrncore_run(const char* arg) {
 #endif
   return r(nrn_nthread, have_gap, nrnmpi_use, nrn_use_fast_imem, arg);
 }
-#else
+
+int (*nrnpy_nrncore_enable_value_p_)();
+/** Return neuron.coreneuron.enable */
+int nrncore_is_enabled() {
+  if (nrnpy_nrncore_enable_value_p_) {
+    int b = (*nrnpy_nrncore_enable_value_p_)();
+    return b;
+  }
+  return 0;
+}
+
+char* (*nrnpy_nrncore_arg_p_)(double tstop);
+/** Run coreneuron with arg string from neuron.coreneuron.nrncore_arg(tstop) */
+int nrncore_psolve(double tstop) {
+  if (nrnpy_nrncore_arg_p_) {
+    char* arg = (*nrnpy_nrncore_arg_p_)(tstop);
+    if (arg) {
+      nrncore_run(arg);
+      free(arg);
+      return 1;
```

Comment:
returning 1 means successful? should be other way around? i.e. 0 means all was good otherwise error?

Reply:
Good point.  I'll review what nrncore_run is returning and pass that back up the stack.  nrncore_run is  no longer being used for nrncore_is_enabled information. The return in case of nrnpy_nrncore_arg_p_ == NULL and arg == NULL should have values that don't conflict with any return value of nrncore_run(arg).  Now I have a nitpick:) why is corenrn_embedded_run in mechanism/mech/enginemech.cpp. It has nothing to do with mechanisms.  Anyway I see
```* @return 1 if embedded mode is used otherwise 0``` instead any error status. Now I worry that coreneuron just exits in case of error.

Reply:
> * @return 1 if embedded mode is used otherwise 0 instead any error status. Now I worry that coreneuron just exits in case of error. 

I think if something is failed (neuron/coreneuron side), there should be some warning/error. But this we could do separately.